### PR TITLE
DAOS-10683 build: Restrict SPDK version to less than 22

### PR DESCRIPTION
**Pull request by tanabarr**
Restrict SPDK version on release/2.2 to avoid any interference when
upgrading to v22.01.1 on master branch.

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

### CODE REVIEW

**Review by johannlombardi — APPROVED** [review]

File: utils/rpms/daos.spec [hunk 1]
```diff
@@ -87,6 +87,7 @@ BuildRequires: protobuf-c-devel
 BuildRequires: lz4-devel
 %endif
 BuildRequires: spdk-devel >= 21.07
+BuildRequires: spdk-devel < 22
```
Comment (brianjmurrell) [inline utils/rpms/daos.spec:90]: While we typically want each dependency on it's own line, a version range can be specified on a single line:

```suggestion
BuildRequires: spdk-devel >= 21.07, spdk-devel < 22
```

That makes it more difficult to miss that it's actually a range, not just a _greater than or equal to_.

File: utils/rpms/daos.spec [hunk 2]
```diff
@@ -176,6 +177,7 @@ to optimize performance and cost.
 Summary: The DAOS server
 Requires: %{name}%{?_isa} = %{version}-%{release}
 Requires: spdk-tools >= 21.07
+Requires: spdk-tools < 22
```
Comment (brianjmurrell) [inline utils/rpms/daos.spec:180]:
```suggestion
Requires: spdk-tools >= 21.07, spdk-tools < 22
```

**Review by brianjmurrell — CHANGES_REQUESTED** [review]
This range pinning really should be expressed in the `debian/control` also:

https://github.com/daos-stack/daos/blob/e69cb998e4034d5411163bf1cc0485b9e999f58b/debian/control#L29

```patch
-  libspdk-dev, 
+  libspdk-dev (>=21.07), libspdk-dev (<<22)
```

It's definitely an oversight that it was not added in 5260802 or any update subsequent to that.

File: debian/control [hunk 3]
```diff
@@ -21,12 +21,12 @@ Build-Depends: debhelper (>= 10),
                libfuse3-dev,
                libprotobuf-c-dev,
                libjson-c-dev,
-               dpdk-dev,
+               dpdk-dev (<<21.11),
```
Comment (brianjmurrell) [inline debian/control:24]:
```suggestion
```

**Review by brianjmurrell — CHANGES_REQUESTED** [review]
Missing `debian/changes` update.

Please remember to let me resolve conversations that I start.  It helps me keep track the issues I raise in the many PRs that I am asked to review.

File: debian/control (hunk 3 above)
Comment (brianjmurrell) [reply debian/control:24]: > done

Is it a GitHub anomaly or is this not actually removed as suggested?

**Review by brianjmurrell — CHANGES_REQUESTED** [review]
Per previous comments.

**Comment by brianjmurrell** [comment]
Looks like we don't have the equivillent of https://github.com/daos-stack/spdk/blob/fd3be59285e44b76d355e404162cfaf5088fe2c3/spdk.spec#L79 in https://github.com/daos-stack/spdk/blob/fd3be59285e44b76d355e404162cfaf5088fe2c3/debian/control#L24-L26 which is causing the most recent DEB build failure.

**Comment by tanabarr** [comment]
> Looks like we don't have the equivillent of https://github.com/daos-stack/spdk/blob/fd3be59285e44b76d355e404162cfaf5088fe2c3/spdk.spec#L79 in https://github.com/daos-stack/spdk/blob/fd3be59285e44b76d355e404162cfaf5088fe2c3/debian/control#L24-L26 which is causing the most recent DEB build failure.

So this requires a bump to the release version of our SPDK v21.07 to fix, correct?

**Comment by brianjmurrell** [comment]
> So this requires a bump to the release version of our SPDK v21.07 to fix, correct?

While my comment is technically correct, it doesn't seem reasonable to try to fix history at this point.  So while it would be most correct to update the spdk being used for _release/2.x_ branches, to have spdk-dev require dpdk-dev like the RPMs do, it's not practical, so let's just continue to have the `debian/control` have `dpdk-dev` in it's `Build-Depends:`.  Let's leave the `libisal-dev` in there as that is technically correct, but let's put back the `dpdk-dev (<<21.11)` for the `release/2.x` PRs.  Apologies for all of the confusion.  Just too much going on at once to keep it all straight unfortunately.

**Review by brianjmurrell — APPROVED** [review]

**Review by mjmac — APPROVED** [review]

**Pull request merged**